tools/lib/provision.py

```diff
@@ -32,6 +32,17 @@
 is_travis = 'TRAVIS' in os.environ
 is_circleci = 'CIRCLECI' in os.environ
 
+# Check that the python version is greater than the
+# minimum version we support.
+MIN_PY_SUPPORTED_RELEASE = (3, 6)
+if sys.version_info < MIN_PY_SUPPORTED_RELEASE:
+    print(FAIL + "Error: Unsupported python version {} found.".format(sys.version[:5]) + ENDC)
+    print("Atleast python version {}.{} is required to run provision.".format(
+        MIN_PY_SUPPORTED_RELEASE[0],
+        MIN_PY_SUPPORTED_RELEASE[1]
+    ))
+    sys.exit(1)
```

Comment:
I think we only want this to run after verifying the OS release, if we do it at all; upgrading Python on Ubuntu Xenial as the thing the developer thinks hey should fix won't be a good experience.

And one can't get Python 3.5 on any of the platforms we now support, so I think after moving this, it will never run, so we probably shouldn't do it.

Reply:
Agreed!